### PR TITLE
Convert Rook NFS to a Helm chart and install it as a Slurm chart dependency

### PR DESCRIPTION
**Pull request by sd109**

### CODE REVIEW

File: slurm-cluster-chart/values.yaml [hunk 1]
```diff
@@ -1,12 +1,52 @@
...
   mountPath: /home
-  claimName: rook-nfs-pv-claim
+  # The name of a Read-Write-Many StorageClass to use for
+  # the persistent volume which is shared across Slurm nodes
+  # Note: If using the default value then you must set
+  # rooknfs.enabled = true below to ensure that Rook NFS is 
+  # installed on the cluster as a dependency of this Slurm
+  # chart. If you are using a separate RWM StorageClass, then
+  # set rooknfs.enabled = false
+  storageClassName: &storageclassname slurm-rook-nfs
```
Comment (sjpb) [inline slurm-cluster-chart/values.yaml:28]: After discussion, lets ditch the yaml anchor here and just set a default 
```suggestion
  storageClassName: slurm-rook-nfs
```

File: slurm-cluster-chart/values.yaml [hunk 2]
```diff
@@ -1,12 +1,52 @@
...
+  claimName: slurm-shared-storage
+  # Capacite of the R-W-M volume
+  capacity: &capacity 10Gi
+
+
+# Values to be passed to the rook-nfs sub-chart
+# See rook-nfs sub-chart for full set of available config values
+rooknfs:
+  enabled: true
+  storageClassName: *storageclassname
```
Comment (sjpb) [inline slurm-cluster-chart/values.yaml:39]: ... and then explain, but allow people to call the rook sc something else if they "need" to:
```suggestion
  storageClassName: slurm-rook-nfs # NB this must match storage.storageClassName when using rook
```

File: slurm-cluster-chart/values.yaml [hunk 3]
```diff
@@ -1,12 +1,52 @@
...
+  # Note: If using the default value then you must set
+  # rooknfs.enabled = true below to ensure that Rook NFS is 
+  # installed on the cluster as a dependency of this Slurm
+  # chart. If you are using a separate RWM StorageClass, then
+  # set rooknfs.enabled = false
+  storageClassName: &storageclassname slurm-rook-nfs
+  # Name for the R-W-M volume to provision
+  claimName: slurm-shared-storage
+  # Capacite of the R-W-M volume
+  capacity: &capacity 10Gi
```
Comment (sjpb) [inline slurm-cluster-chart/values.yaml:32]: Leave the yaml anchor, but explain
```suggestion
  capacity: &capacity 10Gi   # NB yaml anchor used so this value is also set for `rooknfs.storageCapacity` if necessary.
```

File: slurm-cluster-chart/values.yaml [hunk 4]
```diff
@@ -1,12 +1,52 @@
...
+  storageClassName: *storageclassname
+  # Name for the NFSServer resource created by Rook
+  serverName: rook-nfs
+  # Capacity for the backing Read-Write-*Once* volume 
+  # than Rook will create to provide the actual storage to 
+  # the NFS server. Since we're using the Rook NFS in a 
+  # slightly unconventional way here, we just want to anchor
+  # this value to the requested storage capacity for the RWM
+  # volume specified in storage.capacity
+  storageCapacity: *capacity
```
Comment (sjpb) [inline slurm-cluster-chart/values.yaml:48]:
```suggestion
  storageCapacity: *capacity # size of the underlying RWO PVC used by the NFS server
```

File: slurm-cluster-chart/values.yaml [hunk 5]
```diff
@@ -1,12 +1,52 @@
...
+  storageClassName: &storageclassname slurm-rook-nfs
+  # Name for the R-W-M volume to provision
+  claimName: slurm-shared-storage
+  # Capacite of the R-W-M volume
+  capacity: &capacity 10Gi
+
+
+# Values to be passed to the rook-nfs sub-chart
+# See rook-nfs sub-chart for full set of available config values
+rooknfs:
```
Comment (sjpb) [inline slurm-cluster-chart/values.yaml:37]: Can we expose the storage class name for the "backing" PVC too, so we can use non-default/faster storage if available?

File: slurm-cluster-chart/values.yaml [hunk 6]
```diff
@@ -1,12 +1,52 @@
...
-  login: 1
+login:
+  # Deployment resource name
+  name: login
+  replicas: 1
 
-nfs:
+slurmd:
+  # StatefulSet resource name
+  name: slurmd
```
Comment (sjpb) [inline slurm-cluster-chart/values.yaml:10]:
```suggestion
  name: slurmd # NB this must match NodeName= in slurm-cluster-chart/files/slurm.conf
```

**Review by sjpb — CHANGES_REQUESTED** [review]

File: generate-secrets.sh
Comment (wtripp180901) [inline generate-secrets.sh:1]: Remind me to add this to the helm chart!

File: rooknfs/templates/hooks/pre-delete.yaml
Comment (wtripp180901) [inline rooknfs/templates/hooks/pre-delete.yaml:1]: At some point we will want a dedicated role for this with minimal permissions

**Review by wtripp180901 — APPROVED** [review]

**Review by sjpb — CHANGES_REQUESTED** [review]
Only issue I get is that the removal of no-line-wrapping on generate secrets breaks that script.

File: generate-secrets.sh [hunk 7]
```diff
@@ -1,35 +1,39 @@
...
-kubectl create secret generic database-auth-secret \
+kubectl -n $NAMESPACE create secret generic database-auth-secret \
 --dry-run=client \
 --from-literal=password=$(tr -dc 'A-Za-z0-9' </dev/urandom | head -c 32) \
 -o yaml | \
-kubectl apply -f -
+kubectl -n $NAMESPACE apply -f -
 
-kubectl create secret generic munge-key-secret \
+kubectl -n $NAMESPACE create secret generic munge-key-secret \
```
[the remark this answers is not in the document]
Comment (sd109) [reply generate-secrets.sh:13]: Did you get to the bottom of this? I can't seem to reproduce it.

**Review by sjpb — APPROVED** [review]

**Pull request merged**